packages/flutter/lib/src/cupertino/button.dart

```diff
@@ -212,13 +212,13 @@ class _CupertinoButtonState extends State<CupertinoButton> with SingleTickerProv
     if (_animationController.isAnimating)
       return;
     final bool wasHeldDown = _buttonHeldDown;
-    final TickerFuture ticker = _buttonHeldDown
+    _buttonHeldDown
         ? _animationController.animateTo(1.0, duration: kFadeOutDuration, curve: Curves.easeInOutCubicEmphasized)
-        : _animationController.animateTo(0.0, duration: kFadeInDuration, curve: Curves.easeOutCubic);
-    ticker.then<void>((void value) {
-      if (mounted && wasHeldDown != _buttonHeldDown)
-        _animate();
-    });
+        : _animationController.animateTo(0.0, duration: kFadeInDuration, curve: Curves.easeOutCubic)
+        .then<void>((void value) {
```

Comment:
I might be wrong, but it seems this `then` now only works on the `:`, but not `?`. In my opinion this is a good example of why refactoring that much of code just to meed another style might not worth it

Reply:
@nt4f04uNd you are wrong (not meant negatively, just to answer the "I might be wrong" part 😄).

Reply:
Although I did not go into the full snippet here - I am pretty sure this is what was changed in the latest stable release regarding cascade formatting as well (to make this clear). But I might be wrong 😄

Reply:
```dart
void main() {
  test(true);
  test(false);
  test2(true);
  test2(false);
}

void test(bool value) {
  final future = value ? Future.value('foo') : Future.value('bar');
  future.then((string) { print(string); });
}

void test2(bool value) {
  value
    ? Future.value('foo')
    : Future.value('bar')
  .then((string) { print(string); });
}
```

prints out

```
foo
bar
bar
```